### PR TITLE
SMT-LIB2 QB_BV extensions

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/smt2_solver/basic-bv1/weber_qf_bv.desc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+KNOWNBUG
+weber_qf_bv.smt2
+
+^EXIT=0$
+^SIGNAL=0$
+^unsat$
+--
```
Comment (martin-cs) [inline regression/smt2_solver/basic-bv1/weber_qf_bv.desc:7]: Can you add a version of this with the problematic operators commented out?

**Review by martin-cs — COMMENTED** [review]
Please could you add a test (`.smt2` or other) that is not a KNOWNBUG.  At the moment the changes are listed as dead code.

File: regression/smt2_solver/basic-bv1/weber_qf_bv.desc (hunk 1 above)
Comment (kroening) [reply regression/smt2_solver/basic-bv1/weber_qf_bv.desc:7]: done

File: src/util/bitvector_expr.h [hunk 2]
```diff
@@ -517,8 +517,11 @@ inline extractbits_exprt &to_extractbits_expr(exprt &expr)
 class replication_exprt : public binary_exprt
 {
 public:
-  replication_exprt(const constant_exprt &_times, const exprt &_src)
-    : binary_exprt(_times, ID_replication, _src)
+  replication_exprt(
+    const constant_exprt &_times,
+    const exprt &_src,
+    const typet &_type)
+    : binary_exprt(_times, ID_replication, _src, _type)
```
Comment (tautschnig) [inline src/util/bitvector_expr.h]: In keeping with many (but not all!) other expression constructors: could we instead pass the arguments by value?

**Review by tautschnig — APPROVED** [review]
Looks ok to me.

File: regression/smt2_solver/basic-bv1/weber_qf_bv.desc (hunk 1 above)
Comment (martin-cs) [reply regression/smt2_solver/basic-bv1/weber_qf_bv.desc:7]: Thanks.

**Review by martin-cs — APPROVED** [review]

**Pull request merged**